heudiconv/bids.py

```diff
@@ -0,0 +1,323 @@
+        "TODOs:\n\t- verify and possibly extend information in participants.tsv"
+        "(see for example http://datasets.datalad.org/?dir=/openfmri/ds000208)"
+        "\n\t- fill out dataset_description.json, README, sourcedata/README "
+        "(if present)\n\t- provide _events.tsv file for each _bold.nii.gz with "
+        "onsets of events (see  '8.5 Task events'  of BIDS specification)")
+    create_file_if_missing(op.join(path, 'README'),
+        "TODO: Provide description for the dataset -- basic details about the "
+        "study, possibly pointing to pre-registration (if public or embargoed)")
+
+    # TODO: collect all task- .json files for func files to
+    tasks = {}
+    # way too many -- let's just collect all which are the same!
+    # FIELDS_TO_TRACK = {'RepetitionTime', 'FlipAngle', 'EchoTime',
+    #                    'Manufacturer', 'SliceTiming', ''}
+    for fpath in find_files('.*_task-.*\_bold\.json', topdir=path,
+                        exclude_vcs=True, exclude="/\.(datalad|heudiconv)/"):
+        task = re.sub('.*_(task-[^_\.]*(_acq-[^_\.]*)?)_.*', r'\1', fpath)
+        json_ = load_json(fpath)
+        if task not in tasks:
+            tasks[task] = json_
+        else:
+            rec = tasks[task]
+            # let's retain only those fields which have the same value
+            for field in sorted(rec):
+                if field not in json_ or json_[field] != rec[field]:
+                    del rec[field]
+        # create a stub onsets file for each one of those
+        suf = '_bold.json'
+        assert fpath.endswith(suf)
+        events_file = fpath[:-len(suf)] + '_events.tsv'
+        lgr.debug("Generating %s", events_file)
+        with open(events_file, 'w') as f:
+            f.write("onset\tduration\ttrial_type\tresponse_time")
+    # extract tasks files stubs
+    for task_acq, fields in tasks.items():
+        task_file = op.join(path, task_acq + '_bold.json')
+        lgr.debug("Generating %s", task_file)
+        fields["TaskName"] = ("TODO: full task name for %s" %
+                              task_acq.split('_')[0].split('-')[1])
+        fields["CogAtlasID"] = "TODO"
+        with open(task_file, 'w') as f:
+            f.write(json_dumps_pretty(fields, indent=2, sort_keys=True))
+
+
+def tuneup_bids_json_files(json_files):
+    """Given a list of BIDS .json files, e.g. """
+    if not json_files:
+        return
+    # Harmonize generic .json formatting
+    for jsonfile in json_files:
+        json_ = load_json(jsonfile)
+        # sanitize!
+        for f1 in ['Acquisition', 'Study', 'Series']:
+            for f2 in ['DateTime', 'Date']:
+                json_.pop(f1 + f2, None)
+        # TODO:  should actually be placed into series file which must
+        #        go under annex (not under git) and marked as sensitive
+        # MG - Might want to replace with flag for data sensitivity
+        # related - https://github.com/nipy/heudiconv/issues/92
```

Comment:
"random" shifting of the Date should be done per dataset (so not exactly random - subsequent calls should reuse the same offset which shouldn't be shared later), and could be a nice (optional) feature indeed.  Let's keep this comment in mind, so later on some proper "--anonimize-dates" could get other values (e.g. "random-offset") not only the current "strip" ;)